worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,205 @@
+        else:
+            return 0
+
+    # Helper function to reuse some nasty if/else trees.
+    def item_type_helper(self, item) -> (int, ItemClassification):
+        # Make 101 Power Cells.
+        if item in range(jak1_id, jak1_id + Scouts.fly_offset):
+            classification = ItemClassification.progression_skip_balancing
+            count = 101
+
+        # Make 7 Scout Flies per level.
+        elif item in range(jak1_id + Scouts.fly_offset, jak1_id + Specials.special_offset):
+            classification = ItemClassification.progression_skip_balancing
+            count = 7
+
+        # Make only 1 of each Special Item.
+        elif item in range(jak1_id + Specials.special_offset, jak1_id + Caches.orb_cache_offset):
+            classification = ItemClassification.progression
+            count = 1
+
+        # Make only 1 of each Move Item.
+        elif item in range(jak1_id + Caches.orb_cache_offset, jak1_id + Orbs.orb_offset):
+            classification = ItemClassification.progression
+            count = 1
+
+        # Make N Precursor Orb bundles, where N is 2000 / bundle size.
+        elif item in range(jak1_id + Orbs.orb_offset, jak1_max):
+            classification = ItemClassification.progression_skip_balancing
+            size = self.get_orb_bundle_size()
+            count = int(2000 / size) if size > 0 else 0  # Don't divide by zero!
+
+        # Under normal circumstances, we will create 0 filler items.
+        # We will manually create filler items as needed.
+        elif item == jak1_max:
+            classification = ItemClassification.filler
+            count = 0
+
+        # If we try to make items with ID's higher than we've defined, something has gone wrong.
+        else:
+            raise KeyError(f"Tried to fill item pool with unknown ID {item}.")
+
+        return count, classification
+
+    def create_items(self) -> None:
+        for item_name in self.item_name_to_id:
+            item_id = self.item_name_to_id[item_name]
+
+            # Handle Move Randomizer option.
+            # If it is OFF, put all moves in your starting inventory instead of the item pool,
+            # then fill the item pool with a corresponding amount of filler items.
+            if item_name in self.item_name_groups["Moves"] and not self.options.enable_move_randomizer:
+                self.multiworld.push_precollected(self.create_item(item_name))
+                self.multiworld.itempool += [self.create_item(self.get_filler_item_name())]
+                continue
+
+            # Handle Orbsanity option.
+            # If it is OFF, don't add any orbs to the item pool.
+            # If it is ON, don't add any orb bundles that don't match the chosen option.
+            if (item_name in self.item_name_groups["Precursor Orbs"]
+                and ((self.options.enable_orbsanity == EnableOrbsanity.option_off
+                      or Orbs.to_game_id(item_id) != self.get_orb_bundle_size()))):
```

Comment:
`item_name_groups["Precursor Orbs"]` contains the name of every Orb bundle, even the ones that aren't being using by the current options. This means we need to `continue` past the ones we don't want to create as items. The two conditions we need to meet in order to skip creation are:

1. Orbsanity is totally off, no orbs are added to the item pool, and the orbs in the game behave entirely vanilla. Keep in mind, the OR shortcut logic applies here.
2. The orb bundle we're currently looking at in the loop is not the one we set our options to. We don't want to create this item because (e.g.) the user chose 10 orb bundles, not 5, or 20, or 50.

Reply:
Oh it's iterating over _bundles_ and not individual orbs (well besides like 1-bundles, but those are "bundles") 🤦 That makes sense now, thanks!